src/sync/installation.ts

```diff
@@ -387,6 +386,10 @@ export const markCurrentRepositoryAsFailedAndContinue = async (subscription: Sub
 	const gitHubProduct = getCloudOrServerFromGitHubAppId(subscription.gitHubAppId);
 	statsd.increment(metricTaskStatus.failed, [`type:${nextTask.task}`, `gitHubProduct:${gitHubProduct}`]);
 
+	if (nextTask.task === "repository") {
+		await subscription.update({ syncStatus: SyncStatus.FAILED });
+		return;
```

Comment:
@gxueatlassian - correct, the update to failed alone wont do anything, mostly tidy up.
The return on the next line is what skips the task being fired again.